src/CMakeLists.txt

```diff
@@ -181,11 +181,15 @@ target_include_directories(checkShared
     $<INSTALL_INTERFACE:include>
 )
 
-install(TARGETS check checkShared
-  EXPORT check-targets
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-)
+if (THIS_IS_MAIN_PROJECT)
```

Comment:
If the Check is not the main project then its library is not installed. Does this mean that the main project is not expected to install anything that depends on Check. For example, the main project is not installing its unit tests. (This makes sense, just checking if that is the rationale).

Reply:
Yes. In this matter I have followed the solutions in Catch2 and Googletest.